Convert UI API Layer modules to pluggable ones (#2139)
kyma-project/kyma: components/ui-api-layer/CONTRIBUTING.md

> **NOTE:** These rules mention terms described in the [Terminology](./docs/terminology.md) document.

### Project structure

> - Place all GraphQL types in the `gqlschema` package. Generate them with the `gqlgen.sh` script. If you need any customization, move them to a separate files, customize and include them in the `config.yml` file.

Reviewer: ```suggestion
- Place all GraphQL types in the `gqlschema` package. Generate them with the `gqlgen.sh` script. If you need any customization, move them to separate files, modify, and include them in the `config.yml` file.
```

Follow these rules while you develop new features for this project.

**General implementation rules:**

> - Every domain resolver, if it is not required to run UI API Layer, should be pluggable. It means that it should implement `PluggableModule` interface from the [`module`](./internal/module) package. To see an example implementation, review the [Service Catalog](./internal/domain/servicecatalog) module.

Reviewer: ```suggestion
- Every domain resolver which is not required to run UI API Layer should be pluggable. It means that it should implement the `PluggableModule` interface from the [`module`](./internal/module) package. To see an example implementation, review the [Service Catalog](./internal/domain/servicecatalog) module.
```

**Kubernetes resources:**
- For read only operations, use SharedIndexInformers, a client-side caching mechanism, which synchronize with Kubernetes API. Use them to find and list resources. SharedIndexInformers have different API from IndexInformers, but it is possible to attach multiple event handlers to them to facilitate future modifications.

> - In order to categorize items in cache store, add indexers for SharedIndexInformers in services.

Reviewer: ```suggestion
- To categorize items in the cache store, add indexers for SharedIndexInformers in services.
```
